services/outputhost/extcache.go

```diff
@@ -237,15 +236,28 @@ func (extCache *extentCache) loadReplicaStream(startAddress int64, startSequence
 
 		// First try to start from the already set offset in metadata
 		if startAddress == 0 {
-			// If consumer group wants to start from a timestamp, get the address from the store
-			startFrom := extCache.startFrom
+			// If consumer group wants to start from a timestamp, get the address from the store.
+			// Note: we take into account any 'delay' that is associated with the CG and the 'skip-older'
+			// time that was specified, by offsetting the time appropriately. we apply the 'start-from'
+			// and 'skip-older' to the "visibility time" (enqueue-time + delay) of the message as opposed
```

Comment:
So a "future start from" is technically disallowed -- you cannot create a CG with a start-from beyond a minute into the future (we have checks in frontend create-consumer-group). That said, it is not illegal to have a future start-from associated with an OpenReadStream call from outputhost to storehost -- though this particular code does not enable that scenario; when a 'delay' is specified, the internal start-from value is further pushed into the past by the 'delay' value.